File: src/problems/graph/Graph.java
package problems.graph;

import java.util.ArrayList;
import java.util.Arrays;

public class Graph {
    public static void main(String[] args) {
        var graph = create();

        System.out.println(graph);
    }

    private static ArrayList<ArrayList<Integer>> create() {
        ArrayList<ArrayList<Integer>> adjacencyList = new ArrayList<>();

        for (int i = 1; i <= 8; i++) {
            adjacencyList.add(new ArrayList<>());
        }

        adjacencyList.get(1).addAll(Arrays.asList(2,3));
        adjacencyList.get(2).addAll(Arrays.asList(1,3));
        adjacencyList.get(3).addAll(Arrays.asList(2,4));
        adjacencyList.get(4).addAll(Arrays.asList(1,3,5));
        adjacencyList.get(5).addAll(Arrays.asList(4));
        adjacencyList.get(6).addAll(Arrays.asList(7));
        adjacencyList.get(7).addAll(Arrays.asList(6));

        return adjacencyList;
    }
}

File: src/problems/graph/kotlin/Graph.kt
package problems.graph.kotlin

import java.util.*

fun main(args: Array<String>) {
    val graph = Graph().create()
    println(graph)
}

class Graph {
    fun create(): ArrayList<ArrayList<Int>> {
        val adjacencyList = ArrayList<ArrayList<Int>>()
        for (i in 1..8) {
            adjacencyList.add(ArrayList())
        }
        adjacencyList[1].addAll(listOf(2, 4))
        adjacencyList[2].addAll(listOf(1, 3))
        adjacencyList[3].addAll(listOf(2, 4))
        adjacencyList[4].addAll(listOf(1, 3, 5))
        adjacencyList[5].addAll(listOf(4))
        adjacencyList[6].addAll(listOf(7))
        adjacencyList[7].addAll(listOf(6))

        return adjacencyList
    }

    fun createWithoutCycle(): ArrayList<ArrayList<Int>> {
        val adjacencyList = ArrayList<ArrayList<Int>>()
        for (i in 1..8) {
            adjacencyList.add(ArrayList())
        }
        adjacencyList[1].addAll(listOf(2))
        adjacencyList[2].addAll(listOf(1, 3))
        adjacencyList[3].addAll(listOf(2))

        adjacencyList[6].addAll(listOf(7))
        adjacencyList[7].addAll(listOf(6))

        return adjacencyList
    }
}